src/crypto/CHIPCryptoPALOpenSSL.cpp

```diff
@@ -433,71 +433,116 @@ CHIP_ERROR Hash_SHA1(const uint8_t * data, const size_t data_length, uint8_t * o
     return CHIP_NO_ERROR;
 }
 
-Hash_SHA256_stream::Hash_SHA256_stream() {}
+// For OpenSSL, we store a pointer to the digest context (EVP_MD_CTX) since EVP_MD_CTX is Opaque.
+static_assert(kMAX_Hash_SHA256_Context_Size >= sizeof(void *),
+              "kMAX_Hash_SHA256_Context_Size needs to at least be able to store a pointer");
 
-Hash_SHA256_stream::~Hash_SHA256_stream()
+// Storing a pointer to EVP_MD_CTX in HashSHA256OpaqueContext instead of the actual EVP_MD_CTX structure, as EVP_MD_CTX was made
+// opaque by OpenSSL and is dynamically allocated.
+static inline void set_inner_hash_evp_md_ctx(HashSHA256OpaqueContext * context, EVP_MD_CTX * evp_ctx)
 {
-    Clear();
+    *SafePointerCast<EVP_MD_CTX **>(context) = evp_ctx;
```

Comment:
Is context guaranteed to be properly aligned for this operation?

Reply:
Consider adding a comment that HashSHA256OpaqueContext is aligned at HashSHA256OpaqueContext which is size_t or `sizeof(psa_hash_operation_t)` ... although I am not sure what the latter is.

Should we `static_assert(CHIP_CONFIG_SHA256_CONTEXT_ALIGN >= sizeof(EVP_MD_CTX *))` just in case (not sure if that is fully sane, but it seems reasonable)

Reply:
Ah, SafePointerCast already checks alignment. Never mind then.